src/instructlab/process/process.py

```diff
@@ -285,28 +293,48 @@ def stop_process(local_uuid, remove=True):
     if remove:
         process_registry.processes.pop(local_uuid, None)
     else:
-        process_registry.processes[local_uuid]["done"] = True
+        process_registry.processes[local_uuid]["status"] = ILAB_PROCESS_STATUS.DONE
     save_registry(process_registry=process_registry)
 
 
 def list_processes():
+    """
+    Constructs a list of processes and their statuses. Marks processes as ready for removal if necessary
+
+    Args:
+        None
+
+    Returns:
+        None
+    """
     process_registry = load_registry()
     if not process_registry:
         logger.info("No processes currently in the registry.")
         return
 
     list_of_processes = []
     for local_uuid, entry in process_registry.processes.items():
+        # assume all processes are running and not ready for removal unless status indicates otherwise
+        status = ILAB_PROCESS_STATUS.RUNNING
+        remove = False
         all_pids = [entry["pid"]] + entry["children_pids"]
         if not all_processes_running(all_pids):
-            remove = entry["done"]
-            stop_process(local_uuid=local_uuid, remove=remove)
-            if remove:
+            # if all of our child or parent processes are not running, we should either a. remove this process, or b. mark it ready for removal after this list
+            if "status" in entry and entry["status"] == ILAB_PROCESS_STATUS.DONE:
+                # if this has been marked as done, it has already been listed once after finishing, remove it
+                process_registry.processes.pop(local_uuid, None)
+                save_registry(process_registry=process_registry)
                 continue
+            # if not, list it, but mark it as ready for removal
+            status = ILAB_PROCESS_STATUS.DONE
+            remove = True
         now = datetime.now()
-
         # Calculate runtime
         runtime = now - datetime.fromisoformat(entry.get("start_time"))
+        if "end_time" in entry:
+            runtime = datetime.fromisoformat(
+                entry.get("end_time")
+            ) - datetime.fromisoformat(entry.get("start_time"))
```

Comment:
We can save time by not doing the same operation twice

```suggestion
        start_time = datetime.fromisoformat(entry.get("start_time"))
        runtime = now - start_time
        if "end_time" in entry:
            runtime = datetime.fromisoformat(
                entry.get("end_time")
            ) - start_time
```